setup.py

```diff
@@ -380,8 +366,8 @@ def run(self):
       package_dir = {'':'src'},
       packages=find_packages('src') + ['allmydata.test.plugins'],
       classifiers=trove_classifiers,
-      # We support Python 2.7, and Python 3.7 or later.
-      python_requires=">=2.7, !=3.0.*, !=3.1.*, !=3.2.*, !=3.3.*, !=3.4.*, !=3.5.*, !=3.6.*",
+      # We support Python 3.7 or later.
+      python_requires=">=3.7",
```

Comment:
This range had an open upper-bound previously ... so this is not a regression.  However, how about adding `<3.10` here?  And then someone can do a PR adding 3.10 CI and raise it to `<3.11`.